File: newMyDataStructureAlgorithm/algorithm/permutation.java
import java.util.Scanner;

public class permutation {
	static int N, cnt, ans, a[][], team[];
	static boolean check[];


	static void dfs(int depth,int idx) {
		if (depth == N) {
			for (int i = 0; i < N; i++) {
				System.out.print(team[i]+" ");
			}
			System.out.println();
			return;
		}
		for (int i = idx; i < N; i++) {
			// if (check[i]) {
			// 	continue;
			// }
			check[i] = true;
			team[depth] = i;
			dfs(depth + 1,idx+1);
			check[i] = false;
		}
	}

	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		N = sc.nextInt();
		a = new int[N][N];
		team = new int[N];
		check = new boolean[N];
		ans = 987654321;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				a[i][j] = sc.nextInt();
			}
		}
		dfs(0,0);
		System.out.println(ans);
		sc.close();
	}
}

File: newMyDataStructureAlgorithm/algorithm/permutation.js

const permutation = (n) => {
  let arrayList = [];
	const check = Array(n).fill(false);
  for (let i = 0; i < n; i++) {
    arrayList.push(n + 1);
  }

  const dfs = (depth, index) => {
    if (depth === n) {
      let stringSet = '';
      for (let i = 0; i < n; i++) {
        stringSet += arrayList[i] + ' ';
      }
      console.log(stringSet);
      return;
    }

    for (let i = index; i < n; i++) {
      check[i] = true;
      arrayList[depth] = i;
      dfs(depth + 1, index + 1);
      check[i] = false;
    }
  };

  dfs(0, 0);
};

permutation(5);